packages/backend-function/src/factory.ts

```diff
@@ -265,7 +281,9 @@ class FunctionFactory implements ConstructFactory<AmplifyFunction> {
   };
 }
 
-type HydratedFunctionProps = Required<FunctionProps>;
+type HydratedFunctionProps = Required<FunctionProps> & {
+  layers: FunctionLayerReferences;
+};
```

Comment:
why do we need this? Wouldn't `layers` be automatically be present and required in `HydratedFunctionProps`

Reply:
removed this